action/builder.go

```diff
@@ -139,3 +148,91 @@ func (b *EnvelopeBuilder) Build() Envelope {
 	}
 	return &b.elp
 }
+
+// BuildTransfer loads transfer action into envelope
+func (b *EnvelopeBuilder) BuildTransfer(tx *types.Transaction) (Envelope, error) {
+	if tx.To() == nil {
+		return nil, ErrInvalidAct
+	}
+	b.setEnvelopeCommonFields(tx)
+	tsf, err := NewTransfer(tx.Nonce(), tx.Value(), getRecipientAddr(tx.To()), tx.Data(), tx.Gas(), tx.GasPrice())
+	if err != nil {
+		return nil, err
+	}
+	b.elp.payload = tsf
+	return b.build(), nil
+}
+
+func (b *EnvelopeBuilder) setEnvelopeCommonFields(tx *types.Transaction) {
+	b.elp.nonce = tx.Nonce()
+	b.elp.gasPrice = new(big.Int).Set(tx.GasPrice())
+	b.elp.gasLimit = tx.Gas()
+}
+
+func getRecipientAddr(addr *common.Address) string {
+	if addr == nil {
+		return ""
+	}
+	ioAddr, _ := address.FromBytes(addr.Bytes())
+	return ioAddr.String()
+}
+
+// BuildExecution loads executino action into envelope
+func (b *EnvelopeBuilder) BuildExecution(tx *types.Transaction) (Envelope, error) {
+	b.setEnvelopeCommonFields(tx)
+	exec, err := NewExecution(getRecipientAddr(tx.To()), tx.Nonce(), tx.Value(), tx.Gas(), tx.GasPrice(), tx.Data())
+	if err != nil {
+		return nil, err
+	}
+	b.elp.payload = exec
+	return b.build(), nil
+}
+
+// BuildStakingAction loads staking action into envelope from abi-encoded data
+func (b *EnvelopeBuilder) BuildStakingAction(tx *types.Transaction) (Envelope, error) {
+	stakingAddr, _ := address.FromString(address.StakingProtocolAddr)
```

Comment:
two alternative ways:
1. convert in `init` function
2. convert `tx.TO()` to address.Address, and then compare to `address.StakingProtocolAddr`